Source/ASTextNode2.mm

```diff
@@ -360,115 +366,65 @@ - (NSObject *)drawParametersForAsyncLayer:(_ASDisplayLayer *)layer
   [self _ensureTruncationText];
   ASTextContainer *copiedContainer = [_textContainer copy];
   copiedContainer.size = self.bounds.size;
-  NSMutableAttributedString *mutableText = [self.attributedText mutableCopy] ?: [[NSMutableAttributedString alloc] init];
-  
-  [self prepareAttributedString:mutableText];
   
   return @{
-    @"container": copiedContainer,
-    @"text": mutableText,
-    @"bgColor": self.backgroundColor ?: [NSNull null]
-  };
+           @"container": copiedContainer,
+           @"text": [self l_preparedAttributedText],
+           @"bgColor": self.backgroundColor ?: [NSNull null],
+           // Pass along self as a hack so that we can update _lastUsedLayout.
+           // We'll be careful
+           @"instance" : self
+           };
 }
 
 /**
  * If it can't find a compatible layout, this method creates one.
  *
- * NOTE: Be careful to copy `text` if needed.
+ * NOTE: The `container` you pass into this should be a copy, as it may be
+ * retained by the cache in the event of a cache miss.
```

Comment:
Can't we just copy the container in the case of a cache miss? That would be a huge win in some workflows, if that's the only reason / case for the copy.

Reply:
The problem is that currently the callers of this method already have to create a copy, in order to change the container size. So the copy is free. If I copy inside this method, we create two copies. Does that make sense?